common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +152,614 @@ def test_publish_draft_delete(self):
+    EXPORTED_COURSE_AFTER_DIR_NAME = 'exported_course_after'
+
+    def setUp(self):
+        super(UniversalTestProcedure, self).setUp()
+        self.export_dir = self.EXPORTED_COURSE_BEFORE_DIR_NAME
+        self.root_export_dir = None
+        self.contentstore = None
+        self.store = None
+        self.before_publish = True
+
+    @contextmanager
+    def _create_export_dir(self):
+        """
+        Create a temporary export dir - and clean it up when done.
+        """
+        try:
+            export_dir = mkdtemp()
+            yield export_dir
+        finally:
+            rmtree(export_dir, ignore_errors=True)
+
+    @contextmanager
+    def _setup_test(self, modulestore_builder, course_key):
+        """
+        Create the export dir, contentstore, and modulestore for a test.
+        """
+        with self._create_export_dir() as export_dir:
+            # Construct the contentstore for storing the first import
+            with MongoContentstoreBuilder().build() as test_content:
+                # Construct the modulestore for storing the first import (using the previously created contentstore)
+                with modulestore_builder.build(contentstore=test_content) as test_modulestore:
+                    # Create the course.
+                    course = self._create_course(test_modulestore, course_key)
+                    yield export_dir, test_content, test_modulestore, course
+
+    def _export_if_not_already(self):
+        """
+        Check that the course has been exported. If not, export it.
+        """
+        exported_course_path = os.path.join(self.root_export_dir, self.export_dir)
+        if not (os.path.exists(exported_course_path) and os.path.isdir(exported_course_path)):
+            # Export the course.
+            export_course_to_xml(
+                self.store,
+                self.contentstore,
+                self.course_key,
+                self.root_export_dir,
+                self.export_dir,
+            )
+
+    def _make_deprecated_block_key(self, course_key, block_type, block_id):
+        """
+        Return a block key in the deprecated format.
+        """
+        return r'i4x://{ORG}/{COURSE}/{BLOCK_TYPE}/{BLOCK_ID}'.format(
+            ORG=course_key.org,
+            COURSE=course_key.course,
+            BLOCK_TYPE=block_type,
+            BLOCK_ID=block_id,
+        )
+
+    def _make_block_key(self, course_key, block_type, block_id):
+        """
+        Return a block key in the latest format.
+        """
+        return r'block-v1:{ORG}\+{COURSE}\+{RUN}\+type@{BLOCK_TYPE}\+block@{BLOCK_ID}'.format(
+            ORG=course_key.org,
+            COURSE=course_key.course,
+            RUN=course_key.run,
+            BLOCK_TYPE=block_type,
+            BLOCK_ID=block_id,
+        )
+
+    def _make_block_matching_regex(self, course_key, draft, block_type, **kwargs):
+        """
+        Return a regular expression against which a particular block should match.
```

Comment:
It might be better to right this as a composite matcher that works against the `ElementTree` api, so that you don't have to worry about the possibility of handling odd (but valid) xml encodings in your regex.